src/coreclr/gc/unix/gcenv.unix.cpp

```diff
@@ -421,9 +421,28 @@ void GCToOSInterface::FlushProcessWriteBuffers()
         // Iterate through each of the threads in the list.
         for (mach_msg_type_number_t i = 0; i < cThreads; i++)
         {
-            // Request the threads pointer values to force the thread to emit a memory barrier
-            size_t registers = 128;
-            machret = thread_get_register_pointer_values(pThreads[i], &sp, &registers, registerValues);
+            if (__builtin_available (macOS 10.14, iOS 12, tvOS 9, *))
```

Comment:
Hmm, how is this going to work? Are we planning to ship multiple versions of gc library?

Reply:
Yeah macOS 10.14 and tvOS 9 are already the minimum versions, but for .NET 8 we'll bump to iOS 11 as the minimum so it is the only case where this would run.